Write model with basic boolean constraints

```julia
using JuMP
using MiniZinc
model = Model(() -> MiniZinc.Optimizer{Int}("chuffed"))
@variable(model, x, Bin)
@constraint(model, x := false)
optimize!(model)
```
leads to the following error
```julia
ERROR: AssertionError: length(f.args) > 1
Stacktrace:
  [1] _write_expression(io::IOStream, predicates::Set{String}, variables::Dict{MathOptInterface.VariableIndex, NamedTuple{(:name, :lb, :ub, :is_int, :is_bool), Tuple{String, Int64, Int64, Bool, Bool}}}, f::MathOptInterface.ScalarNonlinearFunction)
    @ MiniZinc ~/.julia/packages/MiniZinc/6N0cv/src/write.jl:452
  [2] _write_constraint(io::IOStream, predicates::Set{String}, variables::Dict{MathOptInterface.VariableIndex, NamedTuple{(:name, :lb, :ub, :is_int, :is_bool), Tuple{String, Int64, Int64, Bool, Bool}}}, f::MathOptInterface.ScalarNonlinearFunction, s::MathOptInterface.EqualTo{Bool})
    @ MiniZinc ~/.julia/packages/MiniZinc/6N0cv/src/write.jl:436
  [3] write(io::IOStream, model::MathOptInterface.Utilities.GenericModel{Int64, MathOptInterface.Utilities.ObjectiveContainer{Int64}, MathOptInterface.Utilities.VariablesContainer{Int64}, MiniZinc.ModelFunctionConstraints{Int64}})
    @ MiniZinc ~/.julia/packages/MiniZinc/6N0cv/src/write.jl:526
  [4] #18
    @ ~/.julia/packages/MiniZinc/6N0cv/src/optimize.jl:75 [inlined]
  [5] open(::MiniZinc.var"#18#20"{MiniZinc.Optimizer{Int64}}, ::String, ::Vararg{String, N} where N; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./io.jl:330
  [6] open(::Function, ::String, ::String)
    @ Base ./io.jl:328
  [7] _run_minizinc(dest::MiniZinc.Optimizer{Int64})
    @ MiniZinc ~/.julia/packages/MiniZinc/6N0cv/src/optimize.jl:74
  [8] optimize!(dest::MiniZinc.Optimizer{Int64}, src::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}})
    @ MiniZinc ~/.julia/packages/MiniZinc/6N0cv/src/optimize.jl:210
  [9] optimize!(m::MathOptInterface.Utilities.CachingOptimizer{MiniZinc.Optimizer{Int64}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/2CULs/src/Utilities/cachingoptimizer.jl:316
 [10] optimize!
    @ ~/.julia/packages/MathOptInterface/2CULs/src/Bridges/bridge_optimizer.jl:380 [inlined]
 [11] optimize!
    @ ~/.julia/packages/MathOptInterface/2CULs/src/MathOptInterface.jl:85 [inlined]
 [12] optimize!(m::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.CachingOptimizer{MiniZinc.Optimizer{Int64}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}}}, MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}})
    @ MathOptInterface.Utilities ~/.julia/packages/MathOptInterface/2CULs/src/Utilities/cachingoptimizer.jl:316
 [13] optimize!(model::Model; ignore_optimize_hook::Bool, _differentiation_backend::MathOptInterface.Nonlinear.SparseReverseMode, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ JuMP ~/.julia/packages/JuMP/Gwn88/src/optimizer_interface.jl:457
 [14] optimize!(model::Model)
    @ JuMP ~/.julia/packages/JuMP/Gwn88/src/optimizer_interface.jl:417
 [15] top-level scope
    @ REPL[72]:1
```

The issue comes from `@constraint(model, x := false)`

Replacing it by `fix(x, false, force=true)` leads to a different error
```julia
ERROR: Constraints of type MathOptInterface.VariableIndex-in-MathOptInterface.EqualTo{Float64} are not supported by the solver.

If you expected the solver to support your problem, you may have an error in your formulation. Otherwise, consider using a different solver.
```
which is an issue with JuMP which replaces `false` by `0.0`.

--- a/src/write.jl
+++ b/src/write.jl
@@ -447,7 +447,16 @@ function _write_expression(
     if f.head == :ifelse
         _write_ifelse(io, predicates, variables, f.args)
         return
-    elseif haskey(_INFIX_OPS, f.head)
+    elseif f.head in (:+, :*) && length(f.args) == 1
+        _write_expression(io, predicates, variables, f.args[1])
+        return
+    elseif f.head == :- && length(f.args) == 1
+        print(io, "-(")
+        _write_expression(io, predicates, variables, f.args[1])
+        print(io, ")")
+        return
+    end
+    if haskey(_INFIX_OPS, f.head)
         op = _INFIX_OPS[f.head]
         @assert length(f.args) > 1
         sep = string(" ", op, " ")

--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -1602,6 +1602,34 @@ function test_highs_int_frac_ub()
     return
 end
 
+function test_infix_unary_minus()
+    model = MOI.Utilities.Model{Float64}()
+    x, _ = MOI.add_constrained_variable(model, MOI.Interval(1.0, 3.0))
+    solver = MiniZinc.Optimizer{Float64}("highs")
+    f = MOI.ScalarNonlinearFunction(:-, Any[x])
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+    index_map, _ = MOI.optimize!(solver, model)
+    @test MOI.get(solver, MOI.VariablePrimal(), index_map[x]) ≈ 1.0
+    @test MOI.get(solver, MOI.ObjectiveValue()) ≈ -1.0
+    return
+end
+
+function test_infix_unary_addition()
+    for op in (:+, :*)
+        model = MOI.Utilities.Model{Float64}()
+        x, _ = MOI.add_constrained_variable(model, MOI.Interval(1.0, 3.0))
+        solver = MiniZinc.Optimizer{Float64}("highs")
+        f = MOI.ScalarNonlinearFunction(op, Any[x])
+        MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+        MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+        index_map, _ = MOI.optimize!(solver, model)
+        @test MOI.get(solver, MOI.VariablePrimal(), index_map[x]) ≈ 3.0
+        @test MOI.get(solver, MOI.ObjectiveValue()) ≈ 3.0
+    end
+    return
+end
+
 end
 
 TestMiniZinc.runtests()

